test/image/mocks/image_axis_type.json

```diff
@@ -28,6 +28,7 @@
       "width": 400, "height": 600, "title": {"text": "Image on categorical and log axes"},
       "grid": {"rows": 2, "columns": 1, "pattern": "independent"},
       "xaxis2": {"type": "log"},
-      "yaxis2": {"type": "log"}
+      "yaxis": {"scaleanchor": null},
+      "yaxis2": {"type": "log", "scaleanchor": null}
```

Comment:
`null` isn't a valid `scaleanchor` value. Please remove all the `scalanchor: 'null'` lines from the mocks. If you want to test that `scaleanchor: null` does the right thing, use a jasmine test.

Reply:
... this mean you'll have to update a few baselines, I hope you don't mind.

Reply:
Done in d71c3b345d2eecb7b3087c383f00f5f102b9c2f6

Reply:
Ok, thanks to the new baselines I realised that the `constrain` default is `'range'`.

If I recall correctly, both @emmanuelle and @nicolaskruchten prefered the look of `constrain: 'domain'` for image traces.

Personally, I think `constrain: 'range'` looks just fine and it's probably not worth adding more smart (aka magic) defaults in this case. 

Would it be ok to keep `constrain: 'range'` as default here?

Reply:
Mmm :thinking: I'm not sure. I think `constrain: 'domain'` looks a bit better when a single `image` trace is present. I don't know how I feel about adding some more magic defaults. I'll let @emmanuelle and @nicolaskruchten chime in.

About the smart defaults in `image`, where should we document them? I guess within the trace description in its `index.js` file?

Reply:
> About the smart defaults in `image`, where should we document them? I guess within the trace description in its `index.js` file?

Yep, in the `image/index.js` description would be :ok_hand: